### PR TITLE
Add initial react-query utilities

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

~~Depends on https://github.com/tektoncd/dashboard/pull/2084 due to dependency updates.~~ merged

https://github.com/tektoncd/dashboard/issues/1842

Add a custom hook for fetching a list of namespaced resources
similar to our existing `fetchNamespacedCollection` action creator.
The difference with this is that we won't need action creators,
selectors, or the reducer backing it, all of the caching functionality
is handled internally by react-query while giving us control over
config (e.g. cache time, stale time, retries) if desired.

Add a custom hook to handle websocket events, updating the react-query
cache directly for any matching queries. This is a replacement for
the existing websocket middleware and events handlers we've added with
redux.

These are not yet wired into the application, this will be handled
in another PR.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/main/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: src/api/utils.test.js [hunk 1]
```diff
@@ -143,3 +148,223 @@ describe('checkData', () => {
...
+  const kind = 'TaskRun';
+  it('should handle Created events', () => {
+    const queryClient = getQueryClient();
+    const webSocket = getWebSocket();
+
+    const wrapper = ({ children }) => (
+      <APIWrapper queryClient={queryClient} webSocket={webSocket}>
+        {children}
+      </APIWrapper>
+    );
```
Comment (wingyplus) [inline src/api/utils.test.js:162]: How about we extract it to a function like:

```
// return a wrapper function.
function createWrapper(queryClient, WebSocket) {
  return ({children}) => <APIWrapper queryClient={queryClient} webSocket={webSocket}>{children}</APIWrapper>;
}
```

And then call `createWrapper(queryClient, WebSocket)` in renderHook instead?

Comment (AlanGreene) [reply src/api/utils.test.js:162]: I like this, I've extracted it to a function as you suggested 👍

**Pull request merged**